### PR TITLE
Cpu as device job

### PR DESCRIPTION
**Pull request by stonea**
cpu-as-device is a (relatively) new mode for GPU support set by using `CHPL_GPU=cpu`.  This PR adds a script to conduct nightly testing in this mode and adds various `.skipif`/`.supressif` files to exempt portions of our GPU test suite that are expected (or currently don't) work in cpu-as-device mode.

The tests covered by `.suppressif` are things that I'm somewhat surprised don't currently work so we should do some followup investigation there. In the meantime though, I think locking down what currently works with nightly testing is valuable.

Somewhat tangentially: I also update a function declaration in `chpl-gpu-gen-includes.h` to avoid a compiletime warning due to the prototype having no arguments but instead of having an argument list like `foo(void)` we had it empty like: `foo()`.

### CODE REVIEW

File: test/gpu/native/sharedMemory.skipif
Comment (DanilaFe) [inline test/gpu/native/sharedMemory.skipif:1]: Does this need `==`? the others have `==`.

**Review by DanilaFe — APPROVED** [review]

File: test/gpu/native/nestedForall.suppressif
Comment (e-kayrakli) [reply test/gpu/native/nestedForall.suppressif:4]: Are we getting more launches for cpu-as-device? I might have a theory for why that's the case if that's the case.

File: test/gpu/native/sharedMemory.skipif
Comment (stonea) [reply test/gpu/native/sharedMemory.skipif:1]: Worse than that, this file looks like this file is cruft (there's no `sharedMemory.chpl` test in that dir there's one under `memory/` and in there I have `==`). I've removed this instance of `sharedMemory.skipif`.

File: test/gpu/native/nestedForall.suppressif
Comment (stonea) [reply test/gpu/native/nestedForall.suppressif:4]: > Are we getting more launches for cpu-as-device? 

`nestedForall.chpl:8: error: assert failed - observed 22 launches instead of 2`

Any ideas?

Comment (e-kayrakli) [reply test/gpu/native/nestedForall.suppressif:4]: I'm taking a quick look, but don't hold this PR over that. I am totally fine with suppress/skipifs. If it is an easy fix, I'll queue it up after this PR is merged and we had some nights of clean testing is run.

Comment (stonea) [reply test/gpu/native/nestedForall.suppressif:4]: Good point I've added 

https://github.com/Cray/chapel-private/issues/5064

And I added a note about this issue in the comments of the .supressif files.

**Pull request merged**
